### PR TITLE
Added TimersAsync definition and fixed TriggerAsync class

### PR DESCRIPTION
**Pull request by BlastWind**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2324,9 +2325,9 @@
       "dev": true
     },
     "node_modules/typescript": {
-      "version": "3.8.0-dev.20200207",
-      "resolved": "https://registry.npmjs.org/typescript/-/typescript-3.8.0-dev.20200207.tgz",
-      "integrity": "sha512-JICbjjqDfx99wFJSIrZLKWNsp2tmAi5NYazcNndcTjpwS1sFOtPwX94oJiEAFan75l0MFaBOWnxUMvyAM9M7ew==",
+      "version": "4.7.3",
```
Comment (BlastWind) [inline package-lock.json:2328]: Hmm, maybe I shouldn't mess with the typescript version in package-lock?

File: types/index.d.ts [hunk 2]
```diff
@@ -572,9 +573,22 @@ declare namespace CbServer {
...
-    update(query: Query, changes: object): Promise<object[]>;
-    delete(query: Query): Promise<object>;
+    read(query: AsyncPlatformQuery): Promise<object[]>;
+    update(query: AsyncPlatformQuery, changes: object): Promise<object[]>;
+    delete(query: AsyncPlatformQuery): Promise<object>;
+  }
+
+  interface TimersAsyncCreateOptions extends TimerCreateOptions {
+    name: string;
+    user_id: string;
```
Comment (clarkbynum) [inline types/index.d.ts:583]: According to the [github docs](https://github.com/ClearBlade/native-libraries/blob/master/async.md#timers) user_id and user_type are optional

**Review by clarkbynum — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (clarkbynum) [reply package-lock.json:2328]: I think that's ok since the CI still passes

File: types/index.d.ts [hunk 3]
```diff
@@ -572,9 +573,22 @@ declare namespace CbServer {
...
-    delete(query: Query): Promise<object>;
+    read(query: AsyncPlatformQuery): Promise<object[]>;
+    update(query: AsyncPlatformQuery, changes: object): Promise<object[]>;
+    delete(query: AsyncPlatformQuery): Promise<object>;
+  }
+
+  interface TimersAsyncCreateOptions extends TimerCreateOptions {
+    name: string;
+    user_id?: string;
+    user_type?: number; // user type of user_id (1=dev, 2=user, 3=device)
```
Comment (clarkbynum) [inline types/index.d.ts:584]: You could use a union for this - `user_type?: 1 | 2 | 3`

**Review by clarkbynum — APPROVED** [review]

File: types/index.d.ts (hunk 3 above)
Comment (BlastWind) [reply types/index.d.ts]: Thank you

**Comment by clarkbynum** [comment]
lgtm 👍 feel free to merge. lmk when you want me to push a new version to npm

**Comment by BlastWind** [comment]
> lgtm +1 feel free to merge. lmk when you want me to push a new version to npm

Cool, would be great if you can just push anytime today, I am in no rush.

**Pull request merged**